### PR TITLE
Restore Apollo Client 3.7 `getApolloContext` behaviour

### PR DESCRIPTION
**Pull request by phryneas**
This was triggered over in https://github.com/reduxjs/react-redux/issues/2053

TLDR: AliPay mini widgets deliberately remove anything that is global in their runtime: window, self, global, globalThis, you name it, it's gone.
This doesn't impact the globalThis.DEV, as that can be removed by a compiler, but it does impact situations where we attempt to write to globalThis - so I am rolling back this change to the 3.8 behaviour..

### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [x] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Review by jerelmiller — APPROVED** [review]
Glad you found this before release! Thanks!

**Review by alessbell — APPROVED** [review]
🙌
